OsmAnd/src/net/osmand/plus/GPXUtilities.java

```diff
@@ -181,6 +207,57 @@ public boolean equals(Object obj) {
 			return convert(splitSegments);
 		}
 
+
+		// Track segments are drawn by 'Renderable' objects. A segment can have zero or more renderables,
+		// each of which performs its own point-culling/reduction and display drawing. There are a selction
+		// of renderable types, as defiend in Renderable.RenderType.  To use, call this function to create
+		// a new renderable type, and then attach it to your displayable object in a list or similar.
+
+		// The two parameters' maning varies based upon the type of renderable - see the parameters' usage
+		// in each derived renderable class.
+
+		public Renderable.RenderableSegment addRenderable(OsmandMapTileView view, Renderable.RenderType type,
+														  double param1, double param2) {
+			Renderable.RenderableSegment rs = null;
+			switch (type) {
+				case ORIGINAL: 	// a Ramer-Douglas-Peucer line reduction draw
+					rs = new Renderable.RenderableSegment(type, points, param1, param2);
+					break;
+				case DISTANCE:	// a resample every N metres draw
+					rs = new Renderable.RenderableDot(type, points, param1, param2);
+					break;
+				case CONVEYOR:	// an animating segment draw
+					rs = new Renderable.RenderableConveyor(type, points, param1, param2);
+					Renderable.startScreenRefresh(view, (long) param2);
+					break;
+				case ALTITUDE:	// a colour-banded altitude draw
+					rs = new Renderable.RenderableAltitude(type, points, param1, param2);
+					break;
+				case SPEED:		// a colour-banded speed draw
+					rs = new Renderable.RenderableSpeed(type, points, param1, param2);
+					break;
+
+				default:
+					break;
+			}
+			if (rs != null)
+				renders.add(rs);
+			return rs;
+		}
+
+
+		public void recalculateRenderScales(OsmandMapTileView view) {
+			for (Renderable.RenderableSegment rs : renders)
+				rs.recalculateRenderScale(view);
+		}
+
+		public void drawRenderers(Paint p, Canvas c, RotatedTileBox tb) {
+			for (Renderable.RenderableSegment rs : renders)
```

Comment:
Yes, I am happy to adhere to this coding convention.
I will make the changes and re-upload.
I have also made major changes to more object-oriented in the rendering and resampling.  It is much cleaner - no “type” being passed around anymore.  So it’s good for me to have a chance to update before you seriously consider including my code.



> Just a small remark. Is it possible to put { braces around for, I'm trying to ask people follow java guidelines so it easier for others to follow if it is consistent everywhere.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/osmandapp/Osmand/pull/2379/files/109ca692f100ddb59f0e538b06c32beae5910bc0#r57644348>